shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java

```diff
@@ -371,24 +376,36 @@ public void pushImage(Image image) {
       synchronized (this) {
         toClose = this.image;
         this.image = image;
+        if (image != null) {
+          // Mark that we have a new frame available.
+          markTextureFrameAvailable(id);
+        }
       }
       // Close the previously pushed buffer.
       if (toClose != null) {
         toClose.close();
       }
-      if (image != null) {
-        // Mark that we have a new frame available.
-        markTextureFrameAvailable(id);
-      }
     }
 
     @Override
+    @TargetApi(33)
     public Image acquireLatestImage() {
       Image r;
       synchronized (this) {
         r = this.image;
         this.image = null;
       }
+      if (r != null) {
+        try {
+          SyncFence fence = r.getFence();
+          boolean signaled = fence.awaitForever();
```

Comment:
😢 

Not necessary to do now, but we _might_ be able to get the Vulkan drive to do the waiting via something like https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkExternalFenceHandleTypeFlagBits.html

Reply:
I'm not sure it is possible to convert a SyncFence into a file descriptor (although it is possible to serialize into a Parceable and manually extracting the fd -- just unclear on how stable the parceable format is).

Reply:
Yeah I'm not sure if that is exactly the right API, just that I thought there was something like that